packages/amplify-graphql-default-value-transformer/src/graphql-default-value-transformer.ts

```diff
@@ -79,7 +80,18 @@ const validate = (ctx: TransformerSchemaVisitStepContextProvider, config: Defaul
   validateModelDirective(config);
   validateFieldType(ctx, config.field.type);
   validateDirectiveArguments(config.directive);
-  validateDefaultValueType(ctx, config);
+
+  // Validate the default values only for the DynamoDB datasource.
+  // For RDS, the database determines and sets the default value. We will not validate the value in transformers.
+  const isDynamoDB = isDynamoDBDatasource(ctx, config.object.name.value);
+  if (isDynamoDB) {
+    validateDefaultValueType(ctx, config);
+  }
+};
+
+const isDynamoDBDatasource = (ctx: TransformerSchemaVisitStepContextProvider, modelName: string): boolean => {
```

Comment:
nit: This is something we could re-use across transformers if we put this in the transformer core.